packages/flutter_app_badger/tizen/src/tizen_badge.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "tizen_badge.h"
+
+#include "log.h"
+
+bool TizenBadge::Initialize() {
+  char *app_id = nullptr;
+  int ret = app_get_id(&app_id);
+  if (ret != APP_ERROR_NONE) {
+    LOG_ERROR("app_get_id() failed with error %d.", ret);
+    return false;
+  }
+  app_id_ = app_id;
+  free(app_id);
+
+  ret = badge_add(app_id_.c_str());
+  if (ret == BADGE_ERROR_NOT_SUPPORTED) {
+    LOG_DEBUG("Badge is not supported.");
+    is_supported_ = false;
+  } else if (ret != BADGE_ERROR_NONE && ret != BADGE_ERROR_ALREADY_EXIST) {
+    LOG_ERROR("badge_add() failed with error %d.", ret);
+    return false;
+  } else {
```

Comment:
I believe we can erase this and make `badge_add` call only be in `UpdateBadgeCount`, so that `UpdateBadgeCount` and `RemoveBadge` be symmetric.

Reply:
I think the current logic is fine.

Reply:
Alright.